### PR TITLE
CE-3502 Change Upvote to Nice work!

### PR DESCRIPTION
**Pull request by kamilkoterba**
**Merge after #2213**
## Links

* http://wikia-inc.atlassian.net/browse/CE-3502

## Description

Adjusting layouts to new version with Nice work instead of Upvote

## Reviewers

@Wikia/community-engineering, @Wikia/x-wing

### CODE REVIEW

File: front/main/app/components/article-diff.js [hunk 1]
```diff
@@ -3,17 +3,30 @@ import Ember from 'ember';
 export default Ember.Component.extend({
 	classNames: ['diff-page'],
 	currentUser: Ember.inject.service(),
+	currentUserUpvoteId: Ember.computed('model.upvotes', 'currentUser.userId', function () {
+		const upvotes = this.get('model.upvotes'),
+			currentUserUpvote = upvotes ? upvotes.filterBy(
```
Comment (kamilkoterba) [inline front/main/app/components/article-diff.js]: Changed to findBy in #2213

File: front/main/app/controllers/article-diff.js [hunk 2]
```diff
@@ -125,6 +125,35 @@ export default Ember.Controller.extend(
 			},
 
 			/**
+			 * Send info to server that user downvoted a revision
+			 * @param {int} upvoteId ID of upvote record to remove
+			 * @param {int} userId user ID who made an edit
+			 * @returns {void}
+			 */
+			downvote(upvoteId) {
```
Comment (kamilkoterba) [inline front/main/app/controllers/article-diff.js]: There's a separate task to move upvotes and downvotes functionality to service https://wikia-inc.atlassian.net/browse/CE-3542

File: front/main/app/models/article-diff.js [hunk 3]
```diff
@@ -87,6 +88,40 @@ const ArticleDiffModel = Ember.Object.extend({
...
 		});
+	},
+
+	/**
+	 * Send info to server that user downvoted a revision
+	 * @param {int} upvoteId ID of upvote record to remove
+	 * @param {int} userId user ID who made an edit
+	 * @returns {Ember.RSVP.Promise}
+	 */
+	downvote(upvoteId) {
```
Comment (kamilkoterba) [inline front/main/app/models/article-diff.js]: There's a separate task to move upvotes and downvotes functionality to service https://wikia-inc.atlassian.net/browse/CE-3542

File: front/main/app/components/article-diff.js [hunk 4]
```diff
@@ -3,17 +3,30 @@ import Ember from 'ember';
...
-		const upvotes = this.get('model.upvotes');
-
-		return upvotes && upvotes.some((upvote) => {
-			return parseInt(upvote.from_user, 10) === this.get('currentUser.userId');
-		});
-	}),
-	classUpvote: Ember.computed('upvoted', function () {
-		return this.get('upvoted') ? 'diff-page__upvote--upvoted' : 'diff-page__upvote';
+	upvoted: Ember.computed.bool('currentUserUpvoteId'),
+	buttonVoteClass: Ember.computed('upvoted', function () {
```
Comment (lukaszkonieczny) [inline front/main/app/components/article-diff.js]: What do you think about using `classNameBindings` ?

File: front/common/public/locales/en/recent-wiki-activity.json [hunk 5]
```diff
@@ -2,23 +2,19 @@
   "main": {
     "edited": "edited",
     "dont-show-again": "Don't show me this again",
+    "error": "Sorry, something went wrong. Please try again.",
```
Comment (lukaszkonieczny) [inline front/common/public/locales/en/recent-wiki-activity.json]: Should we change error message key here: https://github.com/Wikia/mercury/pull/2226/files#diff-b1a628e75b90a419288888a4eeea0f48R76 ?

File: front/main/app/styles/_settings.scss [hunk 6]
```diff
@@ -1,9 +1,15 @@
 // Use Wikia style guide settings
 @import '../../bower_components/wikia-style-guide/src/scss/lib/settings';
 
+$appreciate-activated-color: $success-color;
 $auth-content-background-color: $color-white;
 $body-background: $color-blue-dark;
 $button-color: $color-blue-light;
+// TODO move these button colors to styleguide
```
Comment (lukaszkonieczny) [inline front/main/app/styles/_settings.scss]: Do we have a ticket for this?

File: front/main/app/styles/component/_article-diff.scss [hunk 7]
```diff
@@ -94,33 +94,61 @@ $legend-line-height: $type-smallest;
...
 	border: solid $color-gray-light;
 	border-width: 5px 0 0;
 	bottom: 0;
 	box-shadow: 0px -20px 30px 10px rgba(255,255,255,0.68);
-	color: $anchor-font-color;
-	display: block;
+	position: fixed;
+	width: 100%;
+	z-index: $z-site-head;
+	overflow: hidden;
```
Comment (lukaszkonieczny) [inline front/main/app/styles/component/_article-diff.scss]: alphabetical order

File: front/main/app/components/article-diff.js (hunk 4 above)
Comment (kamilkoterba) [inline front/main/app/components/article-diff.js]: `classNameBindings` adds class for component in this case computed field works best to keep the BEM convention.

Comment (lukaszkonieczny) [inline front/main/app/components/article-diff.js]: :+1:

File: front/main/app/styles/_settings.scss (hunk 6 above)
Comment (kamilkoterba) [inline front/main/app/styles/_settings.scss]: https://wikia-inc.atlassian.net/browse/CE-3548

File: front/common/public/locales/en/recent-wiki-activity.json (hunk 5 above)
Comment (kamilkoterba) [inline front/common/public/locales/en/recent-wiki-activity.json]: yes. https://github.com/Wikia/mercury/pull/2226/commits/1ef797d40c980a056e27b6f50c3a40db22a2918b

**Pull request merged**